doc/man1/flux-jobs.adoc

```diff
@@ -0,0 +1,100 @@
+// flux-help-include: true
+FLUX-JOBS(1)
+==============
+:doctype: manpage
+
+
+NAME
+----
+flux-jobs - list jobs submitted to Flux
+
+
+SYNOPSIS
+--------
+*flux* *jobs* ['OPTIONS']
+
+
+DESCRIPTION
+-----------
+flux-jobs(1) is used to list jobs run under Flux.  By default only
+pending and running jobs for the current user are listed.  Additional
+jobs and information can be listed using options listed below.
+
+
+OPTIONS
+-------
+*-a*::
+List all jobs of the current user, including completed jobs is in the
+inactive state.  Equivalent to specifying
+'--state=pending,running,inactive'.
+
+*-A*::
+List all jobs from all users, including completed jobs is in the inactive state.
+Equivalent to specifying '--state=pending,running,inactive --userid=all'.
+
+*--suppress-header*::
+For default output, do not output column headers.
+
+*-u, --userid*'=UID'::
+List jobs for a specific userid.  Specify 'all' for all users.
+
+*-c, --count*'=N'::
+Limit output to N jobs (default 1000)
+
+*-s, --states*'=STATES'::
```

Comment:
I see your point. `--lists`? `--categories`?  I'll see if something better comes to mind tonight.

Reply:
Since this is a porcelain tool, maybe it doesn't matter that the states presented here don't match the flux internal job states? We should use whatever makes sense for the tool itself. That being said, for consistency it might be nice for the names of states in the `--states` option to match the states presented in the `STATE` column in the tool's output.

Reply:
It just occurred to me, `--states` on the command line refers to the job list the job is on (pending, running, inactive), but `STATE` column refers to the job state (depend, sched, run, cleanup, inactive).  That could be confusing ... hmmm ...

Reply:
To me it seems obvious what the `--states` option does. Maybe "pending" and "running" could be considered virtual states for the purposes of this tool, where `"pending" == "sched,depend"` and `"running" == "run,cleanup"` (then rename "inactive" to "complete" or something). I don't think users should have to be aware of the internal implementation of the `job-info` module to use `flux jobs` (or even that there is a `job-info` module for the casual user).

Another idea would be to make the option even more generic, like a `--filter` or `--include` option. This might make it clearer that the names passed into the option don't necessarily directly correspond to "official" job states.

Reply:
Idea: maybe keep `--states` and add  a small section towards the end of the man page that describes states and virtual states, which can be referenced from the option description up top, e.g.

`--states=STATES`::
...where STATES is a comma-separated list of job states, or virtual job states such as _pending_, _running_ , or _inactive_ - see JOB STATES below.

Such as
> JOB STATES
> Jobs may be observed to pass through five job states in Flux: DEPEND, SCHED, RUN, CLEANUP, and INACTIVE (see Flux RFC 21).   For convenience and clarity, some options accept the following virtual job states:  "pending", an alias for DEPEND,SCHED; "running", an alias for RUN,CLEANUP.

Reply:
That seems like a reasonable solution. We may also want to add "active" as an alias for "pending,running" for completeness.

Reply:
```
 --states=STATES::
...where STATES is a comma-separated list of job states, or virtual job states such as pending, running , or inactive - see JOB STATES below.
```

So would a user be able to specify `--states=DEPEND` or `--states=CLEANUP`?  Just don't have that functionality at the moment.

Reply:
My vote is yes.  Just noted that the C api introduces these virtual states:
```c
enum job_list_flags {
    FLUX_JOB_LIST_ALL = 0,
    FLUX_JOB_LIST_PENDING = 1,
    FLUX_JOB_LIST_RUNNING = 2,
    FLUX_JOB_LIST_INACTIVE = 4,
};
```
It would be better IMHO if the API accepted a mask of real states, and the tool took care of translating the virtual states to a state mask.

I'd be OK with this change coming in a later PR since it seems like this is otherwise pretty close.

Reply:
We could also "canonicalize" the virutal states in the state enum, by adding
```c
typedef enum {
    FLUX_JOB_NEW                    = 1,
    FLUX_JOB_DEPEND                 = 2,
    FLUX_JOB_SCHED                  = 4,
    FLUX_JOB_RUN                    = 8,
    FLUX_JOB_CLEANUP                = 16,
    FLUX_JOB_INACTIVE               = 32,   // captive end state
    /* Virtual states for convenience
    */
    FLUX_JOB_PENDING = (FLUX_JOB_DEPEND | FLUX_JOB_SCHED),
    FLUX_JOB_RUNNING = (FLUX_JOB_RUN | FLUX_JOB_CLEANUP),
    FLUX_JOB_ACTIVE = (FLUX_JOB_PENDING | FLUX_JOB_RUNNING),
} flux_job_state_t;
```

Reply:
> I'd be OK with this change coming in a later PR since it seems like this is otherwise pretty close.

Agreed, this should be for a future PR.